apps/frontend/src/lib/storage/localStorage/user/userData.ts

```diff
@@ -2,10 +2,22 @@ import { UserData } from "@/lib/storage/types";
 import { createUserDataBackup } from "@/lib/backup";
 import { saveBackupAndUpdateStorage } from "../utils";
 import { getUserAndSession } from ".";
+import { getChipAttendance } from "@/lib/chip/attendance";
 
 export const updateUserData = async (userData: UserData): Promise<void> => {
   const { user, session } = await getUserAndSession();
 
+  // If attendance is length 0, it should be backfilled
+  if (userData && userData.attendance && userData.attendance.length === 0) {
```

Comment:
This assumes the backfill will only occur once. Happy to update if attendance can change.